src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,194 @@
+                }
+            }
+            catch (Exception ex)
+            {
+                Close(ex);
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Close(string closeReason)
+        {
+            if (!IsClosed)
+            {
+                IsClosed = true;
+                _ = _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, closeReason, _readerCancellationSource.Token).ConfigureAwait(false);
+                TransportClosed?.Invoke(this, new TransportClosedEventArgs { CloseReason = closeReason });
+                _readerCancellationSource.Cancel();
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+
+            if (_readerCancellationSource != null)
+            {
+                _readerCancellationSource.Cancel();
+                _readerCancellationSource.Dispose();
+            }
+        }
+
+        private void Close(Exception ex)
+        {
+            System.Diagnostics.Debug.WriteLine(ex);
+            Close(ex.ToString());
+        }
+
+        private async Task ConnectAsync()
+        {
+            await _webSocket.ConnectAsync(new Uri(_wsEndpoint), _readerCancellationSource.Token).ConfigureAwait(false);
+            ScheduleTransportTask(ReceiveAsync, _readerCancellationSource.Token);
+        }
+
+        private void ScheduleTransportTask(Func<CancellationToken, Task> func, CancellationToken cancellationToken)
+            => Task.Factory.StartNew(() => func(cancellationToken), cancellationToken, TaskCreationOptions.LongRunning, TaskScheduler.Current);
+
+        private async Task ReceiveAsync(CancellationToken token)
+        {
+            try
+            {
+                if (_webSocket.State == WebSocketState.Closed)
+                {
+                    Close("Closed");
+                }
+                else
+                {
+                    byte[] buffer = new byte[DefaultBufferSize];
+
+                    while (!token.IsCancellationRequested && _webSocket.State == WebSocketState.Open)
+                    {
+                        var stringResult = new StringBuilder();
+
+                        WebSocketReceiveResult result;
+
+                        do
+                        {
+                            result = await _webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), _readerCancellationSource.Token).ConfigureAwait(false);
+
+                            if (result.MessageType == WebSocketMessageType.Close)
+                            {
+                                await
+                                    _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, _readerCancellationSource.Token).ConfigureAwait(false);
+                            }
+                            else
+                            {
+                                var str = Encoding.UTF8.GetString(buffer, 0, result.Count);
```

Comment:
Bug: You can't decode partial strings.

Reply:
What needs to be done? here should I add try/catch?

Reply:
Reading https://en.wikipedia.org/wiki/UTF-8 would be a good start.

Reply:
We do similar for StdIOTransport -> https://github.com/microsoft/playwright-dotnet/blob/87b0ca33ca9849678e256ef3741fa5959f7f0cc8/src/Playwright/Transport/StdIOTransport.cs#L184

Reply:
The message is not partial there. As I said above, you can't decode partial messages.

Reply:
One approach that I can think of accumulating all data in a `byte` list i.e it will have data from remote & then encoding it with string.

Reply:
Thanks @pavelfeldman for your eyes on this :) I have fixed this.